### ISSUE TITLE
Exporter not compatible with ElasticSearch 5.X

### ISSUE THREAD

**Issue by Crossfire972** [braedon/prometheus-es-exporter#87]
Hi,

We are running ElasticSearch in AWS with the version 5.2 (which is super EOL),

The parsing of the mappings fails because of this field (https://www.elastic.co/guide/en/elasticsearch/reference/5.6/default-mapping.html) as it does not have a `properties` field.

Especially on that record:

```json
{
    "mappings": {
        "_default_": {
            "dynamic":"strict"
        }
    }
}
```

Here is the error that come up when connecting to the ES 5.X:

```shell
[2020-11-05 21:51:43,155] elasticsearch.INFO MainThread GET $HIDDEN_ES_URL/_stats [status:200 request:0.092s]
[2020-11-05 21:51:43,158] prometheus_es_exporter.INFO MainThread Starting server...
[2020-11-05 21:51:43,159] prometheus_es_exporter.INFO MainThread Server started on port 9206
[2020-11-05 21:51:43,264] elasticsearch.INFO MainThread POST $HIDDEN_ES_URL/_all/_search [status:200 request:0.105s]
[2020-11-05 21:51:43,281] elasticsearch.INFO MainThread POST $HIDDEN_ES_URL/logstash-*/_search [status:200 request:0.017s]
[2020-11-05 21:51:57,801] elasticsearch.INFO MainThread POST $HIDDEN_ES_URL/_all/_search [status:200 request:0.038s]
[2020-11-05 21:52:12,781] elasticsearch.INFO MainThread POST $HIDDEN_ES_URL/_all/_search [status:200 request:0.018s]
[2020-11-05 21:52:14,420] elasticsearch.INFO Thread-2 GET $HIDDEN_ES_URL/_cluster/health?level=indices [status:200 request:0.001s]
[2020-11-05 21:52:14,628] elasticsearch.INFO Thread-2 GET $HIDDEN_ES_URL/_nodes/stats [status:200 request:0.205s]
[2020-11-05 21:52:14,640] elasticsearch.INFO Thread-2 GET $HIDDEN_ES_URL/_alias [status:200 request:0.001s]
[2020-11-05 21:52:14,645] elasticsearch.INFO Thread-2 GET $HIDDEN_ES_URL/_mapping [status:200 request:0.004s]
[2020-11-05 21:52:14,646] prometheus_es_exporter.ERROR Thread-2 Error while fetching Indices Mappings.
Traceback (most recent call last):
  File "/usr/src/app/prometheus_es_exporter/__init__.py", line 139, in collect
    metrics = indices_mappings_parser.parse_response(response, self.metric_name_list)
  File "/usr/src/app/prometheus_es_exporter/indices_mappings_parser.py", line 71, in parse_response
    metrics.extend(parse_index(index, data['mappings'], metric=metric))
  File "/usr/src/app/prometheus_es_exporter/indices_mappings_parser.py", line 55, in parse_index
    counts = count_object_fields(type_mappings, counts=counts)
  File "/usr/src/app/prometheus_es_exporter/indices_mappings_parser.py", line 13, in count_object_fields
    for field, mapping in object_mappings['properties'].items():
KeyError: 'properties'
[2020-11-05 21:52:14,673] elasticsearch.INFO Thread-2 GET $HIDDEN_ES_URL/_stats [status:200 request:0.026s]
```

Do you know if it would be trivial to implement the support for ES 5.X ?

Regards

**Comment by braedon** [comment]
Hi @Crossfire972, it looks like we can just skip the `_default_` type when parsing old style mappings. I'll take a look tomorrow.

**Comment by braedon** [comment]
Fixed in [0.13.1](https://github.com/braedon/prometheus-es-exporter/releases/tag/0.13.1)

**Comment by Crossfire972** [comment]
Thanks a bunch
